src/compiler/checker.ts

```diff
@@ -9971,7 +9967,7 @@ namespace ts {
             if (checkType === wildcardType || extendsType === wildcardType) {
                 return wildcardType;
             }
-            const checkTypeInstantiable = maybeTypeOfKind(checkType, TypeFlags.Instantiable);
+            const checkTypeInstantiable = maybeTypeOfKind(checkType, TypeFlags.Instantiable | TypeFlags.GenericMappedType);

```

Comment:
We previously didn't defer resolution when the check or extends type was a generic mapped type. That definitely was not correct.